clang/lib/AST/ASTDumper.cpp

```diff
@@ -357,6 +364,47 @@ void ASTDumper::VisitCompoundStmt(const CompoundStmt *Node) {
   }
 }
 
+void ASTDumper::VisitVarDecl(const VarDecl *D) {
+  dumpBoundsAnnotations(D->getBoundsAnnotations());
+
+  if (D->hasInit())
+    Visit(D->getInit());
+  
+  VisitWhereClause(D->getWhereClause());
+}
+
+void ASTDumper::VisitWhereClause(const WhereClause *WC) {
+  if (!WC || !DumpWhereClauses)
+    return;
+
+  NodeDumper.AddChild([=] {
+    ColorScope Color(OS, ShowColors, StmtColor);
+    OS << "WhereClause";
+    
+    FactListTy Facts = WC->getFacts(); 
+    for (auto *Fact : Facts) {
+      VisitWhereClauseFact(Fact);
+      }
+  }); 
+}
+
+void ASTDumper::VisitWhereClauseFact(const WhereClauseFact *Fact) {
+  if (auto *BF = dyn_cast<BoundsDeclFact>(Fact)) {    
+    NodeDumper.AddChild([=] {
+      ColorScope Color(OS, ShowColors, DeclNameColor);
+      OS << "BoundsFact: " << BF->getVarDecl()->getQualifiedNameAsString();
```

Comment:
I would prefer the text displayed to be same as the type of the Fact.
`BoundsFact` --> `BoundsDeclFact`

Reply:
OK. Thanks.